internal/services/storage/storage_share_resource.go

```diff
@@ -25,13 +32,17 @@ import (
 )
 
 func resourceStorageShare() *pluginsdk.Resource {
-	return &pluginsdk.Resource{
+	r := &pluginsdk.Resource{
 		Create: resourceStorageShareCreate,
 		Read:   resourceStorageShareRead,
 		Update: resourceStorageShareUpdate,
 		Delete: resourceStorageShareDelete,
 
 		Importer: helpers.ImporterValidatingStorageResourceId(func(id, storageDomainSuffix string) error {
+			if strings.HasPrefix(id, "/subscriptions") {
+				_, err := fileshares.ParseShareID(id)
+				return err
+			}
 			_, err := shares.ParseShareID(id, storageDomainSuffix)
 			return err
```

Comment:
I think we should use the `!features.FivePointOhBeta` flag here to steer the behaviour so it's clear what can be removed post 5.0